### ISSUE TITLE
Create geometry for products without Body representation, but with FootPrint and BoundingBox?

### ISSUE THREAD

**Issue by aothms** [IfcOpenShell/IfcOpenShell#85]
For example the excellent sample model:
https://github.com/openBIMstandards/DataSetSchependomlaan/blob/master/Design%20model%20IFC/IFC%20Schependomlaan.ifc

~~~
#278083= IFCBOUNDINGBOX(#278081,343.,795.,2620.);
#278084= IFCSHAPEREPRESENTATION(#198,'Box','BoundingBox',(#278083));
#278086= IFCCARTESIANPOINT((0.,0.));
#278088= IFCCARTESIANPOINT((343.,0.));
#278090= IFCCARTESIANPOINT((343.,795.));
#278092= IFCCARTESIANPOINT((0.,795.));
#278094= IFCPOLYLINE((#278086,#278088,#278090,#278092,#278086));
#278096= IFCGEOMETRICCURVESET((#278094));
#278098= IFCSHAPEREPRESENTATION(#275652,'FootPrint','GeometricCurveSet',
(#278096));
#278100= IFCPRODUCTDEFINITIONSHAPE($,$,(#278084,#278098));
#278104= IFCSPACE('0asyuu9T1AaPR7AmNDL1_c',#25,'3.01',$,$,#278031,
#278100,'mk',.ELEMENT.,.INTERNAL.,$);
~~~

**Comment by Moult** [comment]
Sorry to resurrect this bug, but I just tried to import data which had a bounding box representation and found that IfcOpenShell wouldn't create it.

I also tried tricks like changing the shape representation identifier to "Body" to trick IfcOpenShell into parsing the IfcBoundingBox, to no success.

**Comment by aothms** [comment]
Yes IfcBoundingBox is not implemented (on purpose I guess, because it always in addition to a Body I guess). You can change it into an IfcBlock, which is implemented and basically the same thing. Is this still something we should advance on?

**Comment by Moult** [comment]
@aothms if it isn't too much work, yes please - running into this problem today, actually. It would be good to be able to create shape on IfcBoundingBox.

IfcBlock is similar, but not the same - it has a position, not a point. Not insurmountable, but annoying :)

**Comment by aothms** [comment]
Ok, this issue was slightly different BTW though, that we use the BoundingBox + 2D footprint to create a true Body representation. But sure I will add a setting or sth to include bounding boxes.

**Comment by BIMvoice** [comment]
This is supported on the current `v0.8.0`: geometry is now generated for a product whose only representation is a `BoundingBox`, via the `keep-bounding-boxes` setting.

Verified with an `IfcSpace` carrying just a `Box`/`BoundingBox` representation (no Body):

```python
s = ifcopenshell.geom.settings()
s.set('keep-bounding-boxes', True)
create_shape(s, space)   # OK, 8 vertices (the box)
```

So the bounding-box half of this request is done (opt-in via the setting, which is sensible since most callers don't want box placeholders by default). If the `FootPrint`/`GeometricCurveSet` fallback is still wanted as a separate behaviour that would be worth its own focused issue, but the core ask here (generate geometry from BoundingBox when there's no Body) is available now.
